cpp/Ice/greeterAsync/Chatbot.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "Chatbot.h"
+
+#include <chrono>
+#include <future>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+GreeterServer::Chatbot::~Chatbot()
+{
+    // Wait for all tasks to complete.
+    for (auto& task : _tasks)
+    {
+        task.wait();
+    }
+}
+
+void
+GreeterServer::Chatbot::greetAsync(
+    string name,
+    function<void(string_view)> response,
+    [[maybe_unused]] function<void(std::exception_ptr)> exception,
+    const Ice::Current&)
+{
+    cout << "Dispatching greet request { name = " << name << " }" << endl;
+
+    // Simulate a long-running background operation by using std::async.
+    // Note that we're moving all arguments except current into the lambda expression.
+    _tasks.push_back(std::async(
```

Comment:
It might be worth noting here that this is thread safe since there's only one configured thread pool thread (the default).

Reply:
Fixed